stan/math/rev/core/autodiffstackstorage.hpp

```diff
@@ -57,28 +92,60 @@ struct AutodiffStackSingleton {
     std::vector<size_t> nested_var_alloc_stack_starts_;
   };
 
-  AutodiffStackSingleton() = delete;
   explicit AutodiffStackSingleton(AutodiffStackSingleton_t const &) = delete;
   AutodiffStackSingleton &operator=(const AutodiffStackSingleton_t &) = delete;
 
-  static inline AutodiffStackStorage &instance() {
+  static constexpr inline AutodiffStackStorage &instance() {
+    return
 #ifdef STAN_THREADS
-    thread_local static AutodiffStackStorage instance_;
+        *
```

Comment:
Wouldn't this be easier to understand as `return *instance_`? And the other branch as `return instance_`?

Reply:
Maybe...but I would be ok with it as is, since all the other ifdefs work the same way by modifying a declaration.

Reply:
would it be easier to understand if those pieces also didn't just modify a definition but rather provided the full definition? (that's really a question... I don't know if it's cleaner yet.)

I'm just trying to protect our future selves. Trust me, we're going to be cursing at ourselves for this.

Reply:
I think we can handle it now and in the future. Also...there would not be any fun for my future me or you! The code has a few ifdefs and the logic is always the same which is why I think we are now and then good.

Reply:
I personally find it much easier to read if only the exact difference is highlighted and until you commented I assumed everyone else would feel the same way. Now I am unsure. Is there anything about this in the style guide?